File: mili-server/src/main/java/fun/bm/mili/command/counter/sub/ResetCommand.java
package fun.bm.mili.command.counter.sub;

import io.papermc.paper.command.brigadier.CommandSourceStack;
import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import net.minecraft.server.MinecraftServer;
import org.leavesmc.leaves.command.CommandContext;
import org.leavesmc.leaves.command.LiteralNode;
import org.leavesmc.leaves.util.HopperCounter;

public class ResetCommand extends LiteralNode {

    public ResetCommand() {
        super("reset");
    }

    @Override
    public boolean requires(CommandSourceStack source) {
        return source.getSender().hasPermission("mili.commands.counter.reset");
    }

    @Override
    public boolean execute(CommandContext context) {
        HopperCounter.resetAll(MinecraftServer.getServer(), false);
        context.getSender().sendMessage(Component.text("Counters reset.", NamedTextColor.GREEN));
        return true;
    }
}

File: mili-server/src/main/kotlin/fun/bm/mili/command/counter/sub/ResetCommand.kt
package fun.bm.mili.command.counter.sub
import com.mojang.brigadier.exceptions.CommandSyntaxException
import org.leavesmc.leaves.command.CommandContext
import org.leavesmc.leaves.command.RootNode
import org.leavesmc.leaves.command.SubNode
import org.leavesmc.leaves.util.HopperCounter
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.NamedTextColor

class ResetCommand(parent: RootNode) : SubNode("reset", "mili.commands.counter.reset", parent) {
    override fun execute(context: CommandContext): Boolean {
        HopperCounter.reset()
        context.sender.sendMessage(Component.text("Counters reset.", NamedTextColor.GREEN))
        return true
    }
}